cmd/trace-agent/subcommands/run/command_windows.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+
+package run
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/cmd/trace-agent/subcommands"
+	"github.com/DataDog/datadog-agent/comp/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/runtime"
+	tracecfg "github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/watchdog"
+
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+	"golang.org/x/sys/windows/svc/eventlog"
+
+	"github.com/spf13/cobra"
+)
+
+var elog debug.Log
+
+type RunParams struct {
+	*subcommands.GlobalParams
+
+	// PIDFilePath contains the value of the --pidfile flag.
+	PIDFilePath string
+	// CPUProfile contains the value for the --cpu-profile flag.
+	CPUProfile string
+	// MemProfile contains the value for the --mem-profile flag.
+	MemProfile string
+	// Foreground contains the value for the --foreground flag.
+	Foreground bool
+	// Debug contains the value for the --debug flag.
+	Debug bool
+}
+
+func setOSSpecificParamFlags(cmd *cobra.Command, cliParams *RunParams) {
+	cmd.PersistentFlags().BoolVarP(&cliParams.Foreground, "foreground", "f", false,
+		"runs the agent in the foreground.")
+	cmd.PersistentFlags().BoolVarP(&cliParams.Debug, "debug", "d", false,
+		"runs the agent in debug mode.")
+}
+
+func Start(cliParams *RunParams, config config.Component) error {
+	// Entrypoint here
+
+	if !cliParams.Foreground {
+		isIntSess, err := svc.IsAnInteractiveSession()
+		if err != nil {
+			fmt.Printf("failed to determine if we are running in an interactive session: %v\n", err)
+		}
+		if !isIntSess {
+			runService(cliParams, config)
+			return nil
+		}
```

Comment:
So if we *are* in an interactive session, we don't run the service at all? I don't have a clear grasp on what interactive-mode is or how it's used.